plugins/nodes/src/js/data/NodesResolver.ts

```diff
@@ -0,0 +1,119 @@
+import { Observable, timer, throwError, of } from "rxjs";
+import { RequestResponse } from "@dcos/http-service";
+import { switchMap, map, share } from "rxjs/operators";
+import { makeExecutableSchema } from "graphql-tools";
+
+import Config from "#SRC/js/config/Config";
+
+import {
+  fetchNodesNetwork,
+  NodeNetworkSchema,
+  NodeNetwork,
+  NodesNetwork
+} from "./NodesNetworkClient";
+
+export interface ResolverArgs {
+  fetchNodesNetwork: () => Observable<RequestResponse<NodesNetwork>>;
+  pollingInterval: number;
+}
+
+export interface GeneralArgs {
+  privateIP?: string;
+  privateIPs?: string[];
+}
+
+export interface NodeQueryArgs {
+  privateIP: string;
+}
+
+export interface NodesQueryArgs {
+  privateIPs: string[];
+}
+
+function isNodeQueryArgs(args: GeneralArgs): args is NodeQueryArgs {
+  return (args as NodeQueryArgs).privateIP !== undefined;
```

Comment:
Should we also check if it is different than empty string?

Reply:
We are checking here if it is defined since the resolver would fail if it isn't defined. With an empty string it would still work but not find any matching nodes which is fine, as long as it doesn't fail.